Apps/GatewayApi/src/Models/CreateDelegateInvitationRequest.cs

```diff
@@ -0,0 +1,49 @@
+//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.GatewayApi.Models
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using HealthGateway.Common.Data.Constants;
+
+    /// <summary>
+    /// The CreateDelegateInvitationRequest model.
+    /// </summary>
+    public record CreateDelegateInvitationRequest
+    {
+        /// <summary>
+        /// Gets or sets the nickname.
+        /// </summary>
+        public string Nickname { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the email.
+        /// </summary>
+        public string Email { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the expiry date.
+        /// </summary>
+        public DateOnly ExpiryDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the access for the data sets.
+        /// </summary>
+        [SuppressMessage("Usage", "CA2227:Collection properties should be read only", Justification = "Team decision")]
+        public HashSet<DataSource> DataSources { get; set; } = new();
```

Comment:
Suggest initializing to a static Empty instance instead of allocating a new object.

```
ISet<DataSource> DataSources { get; set; } = ImmutableHashSet<DataSource>.Empty
```

Reply:
It's also possible to do away with the warning instead of supressing it by making the property init-only:

`{ get; init; }`

Reply:
Changed.